node-repository/src/main/java/com/yahoo/vespa/hosted/provision/maintenance/NodeFailer.java

```diff
@@ -192,8 +195,8 @@ else if (hostSuspended(node, activeNodes)) {
 
     /** Returns whether node has any kind of hardware issue */
     static boolean hasHardwareIssue(Node node, NodeRepository nodeRepository) {
-        Node hostNode = node.parentHostname().flatMap(parent -> nodeRepository.nodes().node(parent)).orElse(node);
-        return reasonsToFailParentHost(hostNode).size() > 0;
+        Node host = node.parentHostname().flatMap(parent -> nodeRepository.nodes().node(parent)).orElse(node);
```

Comment:
Consider reusing the node snapshot retrieved in `findActiveFailingNodes()` here.